src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1184,6 +1184,19 @@ public override void VisitMarkupElement(MarkupElementSyntax node)
                             BuildSourceSpanFromNode(node.EndTag), node.EndTag.GetTagNameWithOptionalBang()));
                 }
 
+                if (node.StartTag != null && !_document.Options.SuppressPrimaryMethodBody)
```

Comment:
Without the second condition here, some tests fail. Because some elements are not recognized as tag helper until we get to the second phase of two phase compilation. So this warning gets added during the first phase and breaks the test. Is there are better way to fix it?

Reply:
Hmmm, I think I'd have to understand how/what we're testing in this case. During the declaration phase we wipe away the method body (including the markup). So I guess lets look a the test strategy being used and try to figure this out.

Reply:
Ah I see now that you're checking the condition. I think this is a totally sensible solution - we genuinely don't want this error during the declaration pass 👍